src/App.tsx

```diff
@@ -1,73 +1,117 @@
-import { createBrowserHistory, History } from 'history';
+import { createBrowserHistory } from 'history';
 import * as React from 'react';
 import * as ReactGA from 'react-ga';
 import { IntlProvider } from 'react-intl';
-import { connect, MapStateToProps } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import { Router } from 'react-router';
 import { gaTrackerKey } from './api';
 import { ErrorWrapper } from './containers';
-import { RootState } from './modules';
+import * as mobileTranslations from './mobile/translations';
+import {
+    selectCurrentLanguage,
+    selectMobileDeviceState,
+    setMobileDevice,
+} from './modules';
 import { languageMap } from './translations';
 
-interface AppProps {
-    history: History;
-}
-
-interface ReduxProps {
-    lang: string;
-}
-
 const gaKey = gaTrackerKey();
-const history = createBrowserHistory();
+const browserHistory = createBrowserHistory();
 
 if (gaKey) {
     ReactGA.initialize(gaKey);
-    history.listen(location => {
+    browserHistory.listen(location => {
         ReactGA.set({ page: location.pathname });
         ReactGA.pageview(location.pathname);
     });
 }
 
-type Props = AppProps & ReduxProps;
+/* Mobile components */
+const MobileFooter = React.lazy(() => import('./mobile/components/Footer').then(({ Footer }) => ({ default: Footer })));
+const MobileHeader = React.lazy(() => import('./mobile/components/Header').then(({ Header }) => ({ default: Header })));
 
+/* Desktop components */
 const AlertsContainer = React.lazy(() => import('./containers/Alerts').then(({ Alerts }) => ({ default: Alerts })));
 const CustomizationContainer = React.lazy(() => import('./containers/Customization').then(({ Customization }) => ({ default: Customization })));
 const FooterContainer = React.lazy(() => import('./containers/Footer').then(({ Footer }) => ({ default: Footer })));
 const HeaderContainer = React.lazy(() => import('./containers/Header').then(({ Header }) => ({ default: Header })));
 const SidebarContainer = React.lazy(() => import('./containers/Sidebar').then(({ Sidebar }) => ({ default: Sidebar })));
 const LayoutContainer = React.lazy(() => import('./routes').then(({ Layout }) => ({ default: Layout })));
 
-class AppLayout extends React.Component<Props, {}, {}> {
-    public componentDidMount() {
-        ReactGA.pageview(history.location.pathname);
+
+const handleChangeWindowWidth = (dispatch, isMobileDevice: boolean) => {
+    const height = window.innerHeight;
+    const width = window.innerWidth;
+    let updatedMobileDeviceState = false;
+
+    if ((width && width < 768) || (height && height < 600)) {
+        updatedMobileDeviceState = true;
+    }
+
+    if (isMobileDevice !== updatedMobileDeviceState) {
+        dispatch(setMobileDevice(updatedMobileDeviceState));
```

Comment:
try to not use `dispatch` in base methods. IHMO, enough to use `window.innerHeight` ...

Reply:
thanks, fixed

Reply:
example for using windows size - https://usehooks.com/useWindowSize/

Reply:
it's already done like this, check useSetMobileDevice hook